lib/poison/decoder.ex

```diff
@@ -40,9 +40,13 @@ defmodule Poison.Decode do
   end
 
   defp transform_struct(value, _keys, as, options) do
-    kv = for k <- Map.keys(Map.from_struct(as.__struct__)) do
-      {k, Map.get(value, Atom.to_string(k))}
-    end
+    kv = Map.keys(Map.from_struct(as.__struct__))
+          |> Enum.reduce([], fn key, acc ->
+               case Map.get(value, Atom.to_string(key)) do
+                 nil -> acc
+                 value -> [{key, value} | acc]
+               end
+             end)
```

Comment:
Nifty -- I agree that it'd be great to be able to create the struct in one pass, but I had to convert the struct to a list to make it enumerable. Is there a way to make this better?

```elixir
Enum.into(Map.to_list(as.__struct__), %{}, fn
  ...
```

And, good point on the `null` values overriding the default -- I'll add a test.